### PR TITLE
Implement tag find command

### LINKED ISSUE #15: As an efficient salesman, I can search contacts, sales etc. by tags
... so that I can identify items belonging to a category easily.

### PR DESCRIPTION
**Pull request by Asthenosphere**
Closes #15 

The following screen is the result of the command `tag find 4`
Screenshot 2020-10-12 at 11 34 03 PM

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/tag/FindCommand.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Lists all contacts or sale items associated with the specified tag.
+ */
+public class FindCommand extends Command {
+    public static final String COMMAND_WORD = "tag find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Finds all the contacts or sales items associated with this tag. "
+            + "Note that all contacts or sales associated with this tag "
+            + "will be updated automatically with the updated tag.\n"
```
Comment (sebastiantoh) [inline src/main/java/seedu/address/logic/commands/tag/FindCommand.java:24]: Is this part relevant?

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -114,6 +114,16 @@
      */
     void deleteSaleTag(Tag target);
 
+    /**
+     * Lists all contacts associated with {@code target} tag.
```
Comment (sebastiantoh) [inline src/main/java/seedu/address/model/Model.java]: Is this description accurate? I think the javadocs you have in `AddressBook.java` would be better here! 

Based on the name of the method, I am inclined to think that this method returns a list of contacts. Maybe you want to rename it so it's clearer? Idk. But I think updating the javadocs would be good!

Same for `findBySaleTag`

File: src/main/java/seedu/address/logic/commands/tag/FindCommand.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        if (targetIndex.getOneBased() > contactTagList.size() + saleTagList.size() || targetIndex.getOneBased() < 0) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TAG_DISPLAYED_INDEX);
+        }
+
+        Tag tagToFind;
+        if (targetIndex.getOneBased() > contactTagList.size()) {
+            tagToFind = saleTagList.get(targetIndex.getZeroBased() - contactTagList.size());
+            return new CommandResult(String.format(Messages.MESSAGE_SALES_LISTED_OVERVIEW,
+                    model.findByContactTag(tagToFind)));
```
Comment (sebastiantoh) [inline src/main/java/seedu/address/logic/commands/tag/FindCommand.java]: Should this be be `model.findBySaleTag`?

**Review by sebastiantoh — CHANGES_REQUESTED** [review]
LGTM. Just checking, tagging of sales is not yet implemented right?

File: src/main/java/seedu/address/logic/commands/tag/FindCommand.java (hunk 1 above)
Comment (Asthenosphere) [reply src/main/java/seedu/address/logic/commands/tag/FindCommand.java:24]: Yep, this will be relevant once my next PR is in.

**Review by sebastiantoh — APPROVED** [review]
LGTM!

**Pull request merged**